test/integration/minikube_profile_test.go

```diff
@@ -0,0 +1,53 @@
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"os/exec"
+	"testing"
+
+	"k8s.io/minikube/pkg/minikube/config"
+)
+
+// profileJSON is the output of `minikube profile list -ojson`
+type profileJSON struct {
+	Valid   []config.Profile `json:"valid"`
+	Invalid []config.Profile `json:"invalid"`
+}
+
+func TestMinikubeProfile(t *testing.T) {
+	// 1. Setup two minikube cluster profiles
+	profiles := []string{UniqueProfileName("first"), UniqueProfileName("second")}
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(10))
+	// TODO(@vharsh): Figure out why go vet complains when this is moved into a loop
+	defer CleanupWithLogs(t, profiles[0], cancel)
+	defer CleanupWithLogs(t, profiles[1], cancel)
+	for _, p := range profiles {
+		_, err := Run(t, exec.CommandContext(ctx, Target(), "start", "-p", p))
+		if err != nil {
+			t.Fatalf("test pre-condition failed: minikube start -p " + p + " failed: " + err.Error())
```

Comment:
```suggestion
			t.Fatalf("test pre-condition failed. args %q: %v", rr.Command(), err)
```
This is a good format for output test failures that we use in some other places. Would be nice to keep the consistency. Use this format for the ones below as well if you could, thanks.